core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1242,7 +1259,7 @@ class DAGScheduler(
       stage.latestInfo.completionTime = Some(clock.getTimeMillis())
     } else {
       stage.latestInfo.stageFailed(errorMessage.get)
-      logInfo("%s (%s) failed in %s s".format(stage, stage.name, serviceTime))
+      logInfo(s"$stage (${stage.name}) failed in $serviceTime s due to ${errorMessage.get}")
```

Comment:
the change here is adding `errorMessage`, the rest is just switching to using string interpolation